### PR TITLE
Documentation for v5

### LINKED ISSUE #208: QA the documentation
## Why

We need to ensure all options and tradeoffs are clear to users, as well as the limitations of the default setup.

## What

This includes:
~- payment/setup intents difference~
~- skipping the confirm step~

- [x] authorizing at the ~payment~ confirm step
- [x] implications for non card methods and Europe
- [x] how to migrate to the new extension
- [x] how to integrate with a different frontend

### PR DESCRIPTION
**Pull request by elia**
## Summary

- Includes documentation for https://github.com/solidusio/solidus_stripe/pull/295.
- Fixes #208
- Closes #293 

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [ ] I have written a thorough PR description.
- [ ] I have kept my commits small and atomic.
- [ ] [I have used clear, explanatory commit messages](https://github.com/solidusio/.github/blob/main/CONTRIBUTING.md#writing-good-commit-messages).

The following are not always needed:

- 📖 I have updated the README to account for my changes.
- 📑 I have documented new code [with YARD](https://www.rubydoc.info/gems/yard/file/docs/Tags.md).
- 🛣️ I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- ✅ I have added automated tests to cover my changes.
- 📸 I have attached screenshots to demo visual changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -157,6 +184,16 @@ SolidusStripe.configure do |config|
...
 
+### Customizing the list of available Stripe payment methods
+
+By default, the extension will show all the payment methods that are supported by Stripe in the current currency and for the merchant country.
+
+You can customize the list of available payment methods by overriding the `payment_method_types` option in the `app/views/checkouts/payment/_stripe.html.erb` partial. Please refer to the [Stripe documentation](https://stripe.com/docs/payments/payment-methods) for the full list of supported payment methods.
+
+### Non-card payment methods and "auto_capture"
+
+Solids payment methods are configured with a `auto_capture` option, which is used to determine if the payment should be captured immediately or not. If you intend to use a non-card payment method, it's likely that you'll need to set `auto_capture` to `true` in the payment method configuration. Please refer to the [Stripe documentation](https://stripe.com/docs/payments/payment-methods/integration-options#additional-api-supportability) for more details.
```
Comment (rainerdema) [inline README.md]: Solids -> Solidus

File: README.md [hunk 2]
```diff
@@ -100,13 +109,23 @@ Stripe Payment Method other than "card" on the admin interface, you must include
...
-You can also use [Stripe webhooks](https://stripe.com/docs/webhooks) to trigger
-custom actions in your application.
+Here's the list of events that are supported by default:
 
-First, you need to register the event you want to listen to, both [in
-Stripe](https://stripe.com/docs/webhooks/go-live) and in your application:
+        payment_intent.succeeded
+        payment_intent.payment_failed
+        payment_intent.canceled
+        charge.refunded
```
Comment (waiting-for-dev) [inline README.md:123]: Not against it, as I agree its clearer, but I'm afraid it'll be too easy to forget to add new events here. Your call.

**Review by waiting-for-dev — APPROVED** [review]
:rocket:

**Review by kennyadsl — APPROVED** [review]
👍 Thanks @elia and @rainerdema!

File: README.md (hunk 2 above)
Comment (elia) [reply README.md:123]: Yes, I guess it's the perennial problem of documentation getting out of sync… I added a note to the list of _core events_ in the Event class.

**Pull request merged**